src/github.com/matrix-org/go-neb/services/imgur/imgur.go

```diff
@@ -0,0 +1,288 @@
+	Link        string              `json:"link"`         // The URL link to the album
+	NSFW        *bool               `json:"nsfw"`         // Indicates if the album has been marked as nsfw or not. Defaults to null if information is not available.
+	Topic       string              `json:"topic"`        // Topic of the gallery album.
+	IsAlbum     *bool               `json:"is_album"`     // If it's an album or not
+	Cover       string              `json:"cover"`        // The ID of the album cover image
+	CoverWidth  int                 `json:"cover_width"`  // The width, in pixels, of the album cover image
+	CoverHeight int                 `json:"cover_height"` // The height, in pixels, of the album cover image
+	ImagesCount int                 `json:"images_count"` // The total number of images in the album
+	Images      []imgurGalleryImage `json:"images"`       // An array of all the images in the album (only available when requesting the direct album)
+
+	// ** Unimplemented fields **
+	// account_url	string	The account username or null if it's anonymous.
+	// account_id	integer	The account ID of the account that uploaded it, or null.
+	// privacy	string	The privacy level of the album, you can only view public if not logged in as album owner
+	// layout	string	The view layout of the album.
+	// views	integer	The number of image views
+	// ups	integer	Upvotes for the image
+	// downs	integer	Number of downvotes for the image
+	// points	integer	Upvotes minus downvotes
+	// score	integer	Imgur popularity score
+	// vote	string	The current user's vote on the album. null if not signed in or if the user hasn't voted on it.
+	// favorite	boolean	Indicates if the current user favorited the album. Defaults to false if not signed in.
+	// comment_count	int	Number of comments on the gallery album.
+	// topic_id	integer	Topic ID of the gallery album.
+}
+
+// Imgur gallery search response
+type imgurSearchResponse struct {
+	Data    []json.RawMessage `json:"data"`    // Data temporarily stored as RawMessage objects, as it can contain a mix of imgurGalleryImage and imgurGalleryAlbum objects
+	Success *bool             `json:"success"` // Request completed successfully
+	Status  int               `json:"status"`  // HTTP response code
+}
+
+// Service contains the Config fields for the Imgur service.
+//
+// Example request:
+//   {
+//			"client_id": "AIzaSyA4FD39..."
+//			"client_secret": "ASdsaijwdfASD..."
+//   }
+type Service struct {
+	types.DefaultService
+	// The Imgur client ID
+	ClientID string `json:"client_id"`
+	// The API key to use when making HTTP requests to Imgur.
+	ClientSecret string `json:"client_secret"`
+}
+
+// Commands supported:
+//    !imgur some_search_query_without_quotes
+// Responds with a suitable image into the same room as the command.
+func (s *Service) Commands(client *gomatrix.Client) []types.Command {
+	return []types.Command{
+		types.Command{
+			Path: []string{"imgur", "help"},
```

Comment:
An interesting point here is that this makes it impossible to search on Imgur for the word "help" :) Just a random thought, nothing for you to do.